src/coronawhy_vt/utils/local_processing.py

```diff
@@ -0,0 +1,375 @@
+         "/kaggle/input/CORD-19-research-challenge/comm_use_subset/comm_use_subset/pdf_json/",
+         "/kaggle/input/CORD-19-research-challenge/biorxiv_medrxiv/biorxiv_medrxiv/pdf_json/",
+         "/kaggle/input/CORD-19-research-challenge/custom_license/custom_license/pdf_json/"]
+# TODO Right now this function references an externally declared variable, suggest either moving
+# into a class or make paths an argument
+# Also function name might need to change, a little ambiguous
+
+
+def extract_to_csv(
+        words: list,
+        file_name: str,
+        lemmatized: bool = True,
+        run_title: bool = True,
+        run_abstract: bool = True,
+        run_body: bool = False,
+        replace_dict: dict = None) -> None:
+    """Store aggregate features of vocab words in the main CORD-19 dataset.
+
+    :param words: List of strings representing words of interest
+    :param file_name: Path of the location to store final csv
+    :param lemmatized: If true, perform lemmatized feature extraction
+    :param run_title: If true, features extracted from the title will be included in the aggregate
+    :param run_abstract: If true, features extracted from the abstract will be included in the aggregate
+    :param run_body: If true, features extracted from the body will be included in the aggregate
+    :param replace_dict (dict): {token: replacement_token}
+    """
+    if lemmatized:
+        pull_mentions = pull_mentions_lemmatized
+    else:
+        pull_mentions = pull_mentions_direct
+
+    data_dicts = []
+    if run_title:
+        data_dicts.append(pull_mentions(paths, title_blocks, "title", words, replace_dict))
+    if run_abstract:
+        data_dicts.append(pull_mentions(paths, abstract_blocks, "abstract", words, replace_dict))
+    if run_body:
+        data_dicts.append(pull_mentions(paths, body_blocks, "body", words, replace_dict))
+
+    summed_dictionary = data_dicts[0]
+    for k in data_dicts[0].keys():
+        for d in data_dicts:
+            summed_dictionary[k] = summed_dictionary[k] + d[k]
+
+    dat = pd.DataFrame(summed_dictionary)
+    dat.to_csv(file_name)
+
+
+# TODO change index to be of type int rather than string
+def same_sentence_check(block: str, pos1: str, pos2: str) -> bool:
+    """Check if two words are in the same sentence by looking for sentence delimiters between their starting positions.
+
+    :param block: Block of string text the two words are found in
+    :param pos1: The index of the beginning of word1
+    :param pos2: The index of the beginning of word2
+
+    :return: true if they word1 and word2 are not separated by
+        one of the follwing sentence delimiters ., ;, ?, !, false otherwise
+    """
+    if pos1 < pos2:
+        interstring = block[int(pos1):int(pos2)]
+    else:
+        interstring = block[int(pos2):int(pos1)]
+    sentence_enders = [".", ";", "?", "!"]
+    for s in sentence_enders:
```

Comment:
Could be written more idiomatically with usage of the `any()`
```python
return not any(
    s in interstring
    for s in sentence enders
)
```

or apply de morgan to use `all()`

```python
return all(
    s not in interstring
    for s in sentence enders
)
```

Reply:
Not that it's a big deal for efficiency because it's a list of length 4, but the second one allows for better short-circuit evaluation